src/app/modules/shared-components/components/page-navigator/page-navigator.component.ts

```diff
@@ -10,7 +13,19 @@ export class PageNavigatorComponent {
   @Input() allowEditing = false;
   @Input() allowFullScreen = false;
   @Input() navigationMode = 'nextAndPrev';
+  @Input() groupId?: string;
 
   @Output() edit = new EventEmitter<void>();
   @Output() watch = new EventEmitter<void>();
+  @Output() stopWatch = new EventEmitter<void>();
+
+  isCurrentGroupWatched$ = this.userSessionService.session$.pipe(
+    filter(isNotNullOrUndefined),
+    map(data => data.watchedGroup),
+    filter(isNotNullOrUndefined),
+    map(watchedGroup => this.groupId === watchedGroup.id)
+  );
```

Comment:
That is not correct, this way you'll not go back to false you'll have disable watching (`data.watchedGroup` back to undefined). There should be probably only one map and nothing else as pipe.

Reply:
Yes, i left it because component does re init after start watch or navigation. But it could be forgotten in future if we'll do some refactoring, so I'll fix it

Reply:
Fixed